File: 121.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        if len(prices) == 1:
            return 0
        
        buy = 0
        sell = 1
        maxSell = 0
        for i in (prices[:-1]):
            if prices[buy] < prices[sell]:
                if prices[sell]-prices[buy] > maxSell:
                    maxSell = prices[sell] - prices[buy]
            else:
                buy = sell
            sell+=1
        return maxSell

File: lc-java/121.java
class Solution {
  public int maxProfit(int[] prices) {
    if (prices.length == 1) {
      return 0;
    }

    int buy = 0;
    int sell = 1;
    int maxSell = 0;
    for (int i = 0; i < prices.length - 1; i++) {
      if (prices[buy] < prices[sell]) {
        if (prices[sell] - prices[buy] > maxSell) {
          maxSell = prices[sell] - prices[buy];
        }
      } else {
        buy = sell;
      }
      sell++;
    }

    return maxSell;
  }
}